src/main/java/core/basesyntax/model/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.model.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.handler.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler getOperationHandler(FruitTransaction.Operation operation) {
+        return operationHandlers.get(operation);
```

Comment:
better add a null check and throw an exception if the handler doesn’t exist in the map.